app/controllers/registration_controller.rb

```diff
@@ -0,0 +1,52 @@
+class RegistrationController < ApplicationController
+  def new
+    form_options
+  end
+
+  def create
+    form_options
+    @command = "curl -X GET #{endpoint}?#{registration_args.to_query} #{headers} | bash"
+  end
+
+  private
+
+  def form_options
+    @organizations = Organization.authorized(:view_organizations).select(:id, :name)
+    @locations = Location.authorized(:view_locations).select(:id, :name)
+    @host_groups = Hostgroup.authorized(:view_hostgroups).select(:id, :name)
+    @operating_systems = Operatingsystem.authorized(:view_operatingsystems).select(:id, :title)
+    @smart_proxies = Feature.find_by(name: 'Registration')&.smart_proxies || []
+    @plugin_params = plugin_params
+  end
+
+  def plugin_params
+    Foreman::Plugin.all.map(&:allowed_registration_vars).flatten.compact.uniq
+  end
+
+  def headers
+    jwt = User.current.jwt_token!(expiration: params[:jwt_expiration].to_i.hours.to_i)
+    "-H 'Authorization: Bearer #{jwt}'"
+  end
+
+  def endpoint
+    return global_registration_url if params['smart_proxy'].empty?
+
+    proxy = SmartProxy.authorized(:view_smart_proxies).find_by(id: params[:smart_proxy])
+
+    if proxy
+      "#{proxy.url}/register"
+    else
+      global_registration_url
```

Comment:
Should this instead raise an error? If we got here, user passed a proxy id for which he or she does not have view persmissions on. I think it should cause some exception.